src/core/client/framework/lib/errors/reporter/createReporter.ts

```diff
@@ -0,0 +1,49 @@
+import { StaticConfig } from "coral-common/config";
+
+import { ErrorReporter } from "./reporter";
+import { SentryErrorReporter } from "./sentry";
+
+interface Options {
+  /**
+   * boundary when true will ensure that the boundary element is configured if
+   * the reporter supports it.
+   */
+  boundary?: boolean;
+}
+
+function createReporter({ boundary = false }: Options = {}):
+  | ErrorReporter
+  | undefined {
+  // Parse and load the reporter configuration from the config element on the
+  // page.
+  const element = document.getElementById("config")!;
+  if (!element) {
+    return;
+  }
+
+  // Parse the config from the element text, it should always be a valid JSON
+  // string if the element is available. We should allow the error to bubble if
+  // we are unable to parse it.
+  const config: StaticConfig = JSON.parse(element.innerText);
```

Comment:
this is a bit scary, is there a reason we can't pass these values around like other config values?

Reply:
We want the error reporter setup as _early_ as possible. Otherwise we won't capture network related errors. This is the same mechanism that we pass updates to the public path (See `src/core/build/publicPath.js`).